bigchaindb/models.py

```diff
@@ -365,6 +378,9 @@ def decouple_assets(self):
             the block being the dict of the block with no assets in the CREATE
             transactions.
         """
+        if block_dict is None:
```

Comment:
This seems irrelevant. `block_dict = deepcopy(self.to_dict())` is always called regardless.

Reply:
That is true as we have only one place where `decouple_assets` is called but I tired to generalize and make the interface for `decouple_assets` and `decouple_metadata` similar. We can do it either way

Reply:
That is ok but the logic here is still wrong. `block_dict = deepcopy(self.to_dict())` is always called  and the `if` statement is irrelevant.

Reply:
@r-marques Its my bad, I didn't realize you were talking about [this](https://github.com/bigchaindb/bigchaindb/pull/1812/files#diff-9f1ab5b5daf1b3072e2123b6939bfb07R383)?